src/panel/poi/PoiPanel.jsx

```diff
@@ -232,13 +261,10 @@ export default class PoiPanel extends React.Component {
             !isFromFavorite &&
             (!poiFilters || !poiFilters.category),
           })}
-          renderHeader={!isMobile && backAction !== this.closeAction &&
-            <PanelNav
-              isMobile={isMobile}
-              onGoBack={backAction}
-              goBackText={_('Display all results')}
-            />
-          }
+          renderHeader={<NavHeader isMobile={isMobile} />}
+          floatingItemsLeft={isMobile && [
+            <BackToQwantButton key="back-to-qwant" />,
```

Comment:
Same here, we should check `shouldShowBackToQwant`.
So at the end it will be in four different places, this will be hard to maintain and keep cohesive. Maybe we should try to find a way to factorize that and make it easy to apply to other Panels also? WDYT?

Reply:
Beware, you used `shouldShowBackToQwant` as a boolean instead of a function.